source/common/http/http3/conn_pool.cc

```diff
@@ -7,12 +7,17 @@
 
 #include "common/config/utility.h"
 #include "common/http/http3/quic_client_connection_factory.h"
-#include "common/http/http3/well_known_names.h"
 #include "common/http/utility.h"
 #include "common/network/address_impl.h"
 #include "common/network/utility.h"
 #include "common/runtime/runtime_features.h"
 
+#ifdef ENVOY_ENABLE_QUIC
+#include "common/quic/client_connection_factory_impl.h"
+#else
+#error "http3 conn pool should not be built with QUIC disabled"
```

Comment:
Per moderation's issue with quic header files somehow getting included with the !http3 build, I wonder if we should have a macro for this and include it in all the quic headers right under pragma once.  WDYT?

Reply:
The main issue with that is right now we actually still run the QUIC tests when QUIC is compiled out. Let me see if I can fix this as I agree it would be cleaner.

Reply:
another thought along these lines,what do we think of a macro
QUIC_INCLUDE( "common/quic/client_connection_factory_impl.h")
which translates into this?

having the ifndef inline is more clear, but having a macro does seem easier, and we could optionally late format check nothing in quic is straight up included.

Reply:
Sure I can do this. Like I said elsewhere, there is a much better way of dealing with this once we have Bazel 4.0, which will drop all of the QUIC rules entirely so it simply won't build. I'm happy to do the macro though to make it a bit visually cleaner.

Reply:
OK I tried and failed on this. I don't think there is any way to define a macro that does include or error. We can circle back on this and see if we can make it better later.